src/samplers/gamma.jl

```diff
@@ -162,32 +162,49 @@ end
 # doi:10.1145/358407.358414
 # http://www.cparity.com/projects/AcmClassification/samples/358414.pdf
 
-struct GammaMTSampler <: Sampleable{Univariate,Continuous}
-    d::Float64
-    c::Float64
-    κ::Float64
+# valid for shape >= 1
+struct GammaMTSampler{T<:Real} <: Sampleable{Univariate,Continuous}
+    d::T
+    c::T
+    κ::T
+    r::T
 end
 
 function GammaMTSampler(g::Gamma)
-    d = shape(g) - 1/3
-    c = 1.0 / sqrt(9.0 * d)
+    # Setup (Step 1)
+    d = shape(g) - 1//3
+    c = inv(3 * sqrt(d))
+
+    # Pre-compute scaling factor
     κ = d * scale(g)
-    GammaMTSampler(d, c, κ)
+
+    # We also pre-compute the factor in the squeeze function
+    return GammaMTSampler(promote(d, c, κ, 331//10_000)...)
```

Comment:
On the master branch, an unnecessarily large factor of 0.331 is used in the squeeze function (reference: https://dl.acm.org/doi/10.1145/358407.358414)..